src/lazy/encoder/binary/v1_1/container_writers.rs

```diff
@@ -446,9 +450,45 @@ impl SequenceWriter for BinaryEExpWriter_1_1<'_, '_> {
     }
 }
 
-impl EExpWriter for BinaryEExpWriter_1_1<'_, '_> {
+impl<'top> EExpWriter for BinaryEExpWriter_1_1<'_, 'top> {
+    type ExprGroupWriter<'group>
+        = BinaryExprGroupWriter<'group, 'top>
+    where
+        Self: 'group;
+
     fn write_flex_uint(&mut self, value: impl Into<UInt>) -> IonResult<()> {
         FlexUInt::write(self.buffer, value)?;
         Ok(())
     }
+
+    fn expr_group_writer(&mut self) -> IonResult<Self::ExprGroupWriter<'_>> {
+        todo!("safe binary expression group serialization")
```

Comment:
🪧 I added an API for writing expression groups because having it in text was necessary for testing serialization of compiled macros. However, the text expression group writer is missing important guard rails and the binary one is not fully implemented. These will be implemented in a follow-on PR.